c3po/job/utils/metadata.py

```diff
@@ -0,0 +1,96 @@
+from c3po.db.dao import user, link, song, artist, genre
+from c3po.db.common.base import session_factory
+from datetime import datetime
+from music_metadata_extractor import SongData
+from contextlib import contextmanager
+
+@contextmanager
+def session_scope():
+    session = session_factory()
+    try:
+        yield session
+        session.commit()
+    except Exception as e:
+        raise
+        session.rollback()
+    finally:
+        session.close()
+
+
+def insert(url):
+    with session_scope() as session:
+        data = SongData(url)
+        print(data)
+        new_link = _insert_link(url, session)
+        if(new_link):
+            new_song = _insert_song(data.track, session)
+            _add_song_id(new_song, new_link, session)
+            for artist_data in data.artists:
+                new_artist = _insert_artist(artist_data, session)
+                _insert_artist_song(new_artist, new_song, session)
+
+def _add_song_id(new_song, new_link, session):
+    new_link.song_id = new_song.id
+    session.commit()
```

Comment:
We don't need to add `session.commit()` everywhere, right? If you look at the context manager, it first yields a session and once all the operations are done (i.e. when the scope of the session object is done, it will automatically commit all operations we've done so far within the session - and revert everything in case there's an exception. If this makes sense, please remove the `session.commit()` from all other places. The one usage within the context manager suffices.